src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationSource.cs

```diff
@@ -31,32 +28,32 @@ public IConfigurationProvider Build(IConfigurationBuilder builder)
             try
             {
                 AzureAppConfigurationOptions options = _optionsProvider();
-                ConfigurationClient client;
+                IConfigurationClientProvider clientProvider;
 
-                if (options.Client != null)
+                if (options.ClientProvider != null)
                 {
-                    client = options.Client;
+                    clientProvider = options.ClientProvider;
                 }
                 else if (!string.IsNullOrWhiteSpace(options.ConnectionString))
                 {
-                    client = _configurationClientFactory.CreateConfigurationClient(options.ConnectionString, options.ClientOptions);
+                    clientProvider = new ConfigurationClientProvider(options.ConnectionString, options.ClientOptions);
                 }
-                else if (options.Endpoint != null && options.Credential != null)
+                else if (options.Endpoints != null && options.Credential != null)
                 {
-                    client = _configurationClientFactory.CreateConfigurationClient(options.Endpoint, options.Credential, options.ClientOptions);
+                    clientProvider = new ConfigurationClientProvider(options.Endpoints, options.Credential, options.ClientOptions);
                 }
                 else
                 {
                     throw new ArgumentException($"Please call {nameof(AzureAppConfigurationOptions)}.{nameof(AzureAppConfigurationOptions.Connect)} to specify how to connect to Azure App Configuration.");
                 }
 
-                provider = new AzureAppConfigurationProvider(client, options, _optional);
+                provider = new AzureAppConfigurationProvider(clientProvider, options, _optional);
             }
-            catch (InvalidOperationException e)
+            catch (Exception ex) when (ex is InvalidOperationException || ex is FormatException)
```

Comment:
My understanding is that the exception will only be caught if it satisfies the `when` condition. Isn't it?

Reply:
The `when` condition is good for filtering out things other than type. Type matching is handled with the `catch` itself.

Reply:
That leads to a little bit of duplicated code. I'd argue `when` is better than having duplicated code. Because functionally is not different than having multiple `catch` blocks.